src/coins.cpp

```diff
@@ -1332,32 +1370,37 @@ bool CCoinsViewCache::RevertCeasingScs(const CTxUndo& ceasedCertUndo)
     }
     CCoinsModifier coins = this->ModifyCoins(coinHash);
     LogPrint("cert", "%s():%d - PRE :%s\n", __func__, __LINE__, coins->ToString());
-    unsigned int firstBwtPos = ceasedCertUndo.firstBwtPos;
-
-    const std::vector<CTxInUndo>& outVec = ceasedCertUndo.vprevout;
-    LogPrint("cert", "%s():%d - PRE : outVec.size() = %d\n", __func__, __LINE__, outVec.size());
+    int firstBwtPos = ceasedCertUndo.firstBwtPos;
 
-    for (size_t bwtOutPos = outVec.size(); bwtOutPos-- > 0;)
+    if (firstBwtPos >= 0)
     {
-        LogPrint("cert", "%s():%d - PRE : bwtOutPos= %d\n", __func__, __LINE__, bwtOutPos);
-        if (outVec.at(bwtOutPos).nHeight != 0)
+        const std::vector<CTxInUndo>& outVec = ceasedCertUndo.vprevout;
+        LogPrint("cert", "%s():%d - PRE : outVec.size() = %d\n", __func__, __LINE__, outVec.size());
+
+        for (size_t bwtOutPos = outVec.size(); bwtOutPos-- > 0;)
         {
-            coins->fCoinBase  = outVec.at(bwtOutPos).fCoinBase;
-            coins->nHeight    = outVec.at(bwtOutPos).nHeight;
-            coins->nVersion   = outVec.at(bwtOutPos).nVersion;
-            coins->originScId = outVec.at(bwtOutPos).originScId;
-        } else {
-            LogPrint("cert", "%s():%d - returning false\n", __func__, __LINE__);
-            return false;
-        }
+            LogPrint("cert", "%s():%d - PRE : bwtOutPos= %d\n", __func__, __LINE__, bwtOutPos);
+            if (outVec.at(bwtOutPos).nHeight != 0)
+            {
+                coins->fCoinBase          = outVec.at(bwtOutPos).fCoinBase;
+                coins->nHeight            = outVec.at(bwtOutPos).nHeight;
+                coins->nVersion           = outVec.at(bwtOutPos).nVersion;
+                coins->nBwtMaturityHeight = outVec.at(bwtOutPos).nBwtMaturityHeight;
+            } else {
+                LogPrint("cert", "%s():%d - returning false\n", __func__, __LINE__);
+                return false;
+            }
 
-        if(coins->IsAvailable(firstBwtPos + bwtOutPos))
-            fClean = fClean && error("%s: undo data overwriting existing output", __func__);
-        if (coins->vout.size() < (firstBwtPos + bwtOutPos+1))
-            coins->vout.resize(firstBwtPos + bwtOutPos+1);
-        coins->vout.at(firstBwtPos + bwtOutPos) = outVec.at(bwtOutPos).txout;
+            if(coins->IsAvailable(firstBwtPos + bwtOutPos))
+                fClean = fClean && error("%s: undo data overwriting existing output", __func__);
+            if (coins->vout.size() < (firstBwtPos + bwtOutPos+1))
+                coins->vout.resize(firstBwtPos + bwtOutPos+1);
+            coins->vout.at(firstBwtPos + bwtOutPos) = outVec.at(bwtOutPos).txout;
+        }
     }
 
+    SyncBwtCeasing(ceasedCertUndo.refTx, false);
```

Comment:
This would signal also ceasing reverts that will not be committed to the current state! This must be invoked only in block connect / disconnect as for synctransaction. Add an output parameter list that will be filled withe the unceased sidechains.